src/cascadia/WindowsTerminal/WindowEmperor.cpp

```diff
@@ -438,15 +441,13 @@ void WindowEmperor::HandleCommandlineArgs(int nCmdShow)
     __assume(false);
 }
 
-void WindowEmperor::_dispatchSpecialKey(MSG& msg) const
+void WindowEmperor::_dispatchSpecialKey(const MSG& msg) const
 {
-    const auto hwnd = msg.hwnd;
+    // Each CoreInput window is a child of our IslandWindow.
+    // We can figure out the IslandWindow HWND via GetAncestor().
+    const auto hwnd = GetAncestor(msg.hwnd, GA_ROOT);

```

Comment:
huh, which one did this fix?

Reply:
You can consider this an extra. 😄 When I added this code you asked if we can't figure out the IslandWindow from a CoreInput target and I said no. Turns out we can.